src/SANDECALClustering.cpp

```diff
@@ -0,0 +1,92 @@
+#include "SANDClustering.h"
+
+int clustering(std::string const& input)
+{
+  
+  const char* finname = input.c_str();
+  TFile f(finname, "READ");
```

Comment:
I'd add a check for the existence of the TFile to prevent problems when trying to get the tDigit Tree
something like that 

if (f.IsZombie()){
 std::cerr << "file " << finname << " not found. Process aborted." << std::.endl;
 return -1;
}